lms/lib/courseware_search/lms_result_processor.py

```diff
@@ -73,7 +73,14 @@ def get_display_name(category, item_id):
 
         def get_position_name(section, position):
             """ helper to fetch name corresponding to the position therein """
-            pos = int(position)
+            if position is None:
+                return None
+
+            try:
+                pos = int(position.split('_', 1)[0])
+            except (ValueError, TypeError):
+                pos = 1
```

Comment:
Not really - one is parsing it to get an integer, and the other is parsing it to get it as a string - so it would actually complicate the code more to bundle essentially a one-line item to perform 2 (albeit slightly) different things.